### DISCUSSION TITLE
Problems with compile blue-retro

### DISCUSSION THREAD

**Discussion by parfuar** [darthcloud/BlueRetro#752] [Q&A]
Hi,
I'm trying to compile the BlueRetro code, I need to do some tests 😊.
A used espressif-ide-setup-2.10.0-with-esp-idf-5.0.2 (https://[dl.espressif.com/dl/esp-idf/](https://dl.espressif.com /dl/esp-idf/)) for windows 10.
when trying to compile, my first problem there is no file "brownout_ll.h" (main/startup.c), I tried to use a "brownout_ll.h" from the other project.
My second problem is in startup.c again at line 518 funcion "wdt_hal_context_t rtc_wdt_ctx = RWDT_HAL_CONTEXT_DEFAULT();",
"RWDT_HAL_CONTEXT_DEFAULT()" -> error: invalid initializer.

Probably the RWDT_HAL_CONTEXT_DEFAULT() initializer is done through a library, as the compiler gave an error I commented the lines (as I am doing tests).

Now I see these failures with IRAM. The size is larger than the available size.
"....
[2/4] Linking CXX executable BlueRetro.elfFAILED: BlueRetro.elf
cmd.exe /C "cd . && C:\Espressif\tools\xtensa-esp32-elf\esp-2022r1-11.2.0\xtensa-esp32-elf\bin\xtensa-esp32-elf-g++.exe -mlongcalls -Wno-frame-address  @CMakeFiles\BlueRetro.elf.rsp -o BlueRetro.elf  && cd ."
c:/espressif/tools/xtensa-esp32-elf/esp-2022r1-11.2.0/xtensa-esp32-elf/bin/../lib/gcc/xtensa-esp32-elf/11.2.0/../../../../xtensa-esp32-elf/bin/ld.exe: BlueRetro.elf section `.iram0.text' will not fit in region `iram0_0_seg'
c:/espressif/tools/xtensa-esp32-elf/esp-2022r1-11.2.0/xtensa-esp32-elf/bin/../lib/gcc/xtensa-esp32-elf/11.2.0/../../../../xtensa-esp32-elf/bin/ld.exe: IRAM0 segment data does not fit.
c:/espressif/tools/xtensa-esp32-elf/esp-2022r1-11.2.0/xtensa-esp32-elf/bin/../lib/gcc/xtensa-esp32-elf/11.2.0/../../../../xtensa-esp32-elf/bin/ld.exe: region `iram0_0_seg' overflowed by 2148 bytes
collect2.exe: error: ld returned 1 exit status
ninja: build stopped: subcommand failed.
HINT: The applications static IRAM usage is larger than the available IRAM size.
For more information on how to reduze IRAM usage run 'idf.py docs -sp api-guides/performance/ram-usage.html#optimizing-iram-usage'

Now my doubts. 😊

1 -Is the "brownout_ll.h" library really necessary? If so can you provide?
2- Function "RWDT_HAL_CONTEXT_DEFAULT()" is declares in a library? If yes, can you tell me which library it is?

With these corrections maybe it compiles the code correctly.

Thank you and congratulations for the project

**Comment by darthcloud** [comment]
You need to checkout and follow instruction from
https://github.com/darthcloud/BlueRetroRoot

You need to use same ESP-IDF version as I use.

**Comment by parfuar** [comment]
Hey,
It's already working, thank you

Are the files that import to the board flash?

BlueRetro.bin (inside /build)
bootloader.bin (inside /build/bootloader)
partition-table.bin (inside /build/partition_table)

I became aware of this project because I saw several .bin files (image attached)
Were they made by you? if, yes? Is there any difference between them?

Thanks again 👍

![Capture222](https://github.com/darthcloud/BlueRetro/assets/26558628/3abc63c7-8c89-473f-bee4-a32f13db4ee6)
